<launch>
  <arg name="node_name" default="smoothing" />

  <arg name="image" default="image" doc="The image topic. Should be remapped to the name of the real image topic." />

  <arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
  <arg name="debug_view" default="true" doc="Specify whether the node displays a window to show edge image" />

  <arg name="algorithm_type" default="1" doc="Background Subtraction Algorithm Methods. 0:Mixture-based Gaussian, 1:Mixture-based Gaussian 2, 2:GMG(Godbehere - Matsukawa - Goldberg)" />
Mixture-based Gaussian 

  <!-- smoothing.cpp  -->
  <node name="$(arg node_name)" pkg="opencv_apps" type="background_subtraction" >
    <remap from="image" to="$(arg image)" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="algorithm_type" value="$(arg algorithm_type)" />
  </node>
</launch>
